homework/unique_ptr/unique_ptr.hpp

```diff
@@ -3,18 +3,19 @@
 template<typename T>
 class unique_ptr {
 public:
-    unique_ptr() = default;
-    unique_ptr(T* pointer) : pointer_(pointer) {}
-    unique_ptr(unique_ptr& ptr);
+    unique_ptr(T* pointer = nullptr) : pointer_(pointer) {}
+    unique_ptr(unique_ptr& ptr) = delete;
     unique_ptr(unique_ptr&& ptr);
+    unique_ptr& operator=(const unique_ptr& ptr) = delete;
+    unique_ptr& operator=(unique_ptr&& ptr);
     ~unique_ptr();
     T& operator*() const { return *pointer_; }
     T* operator->() const { return pointer_; }
     T* get() const { return pointer_; }
     T* release();
     void reset(T* new_ptr = nullptr);
 private:
-    T* pointer_ = nullptr;
+    T* pointer_;
```

Comment:
```suggestion
    T* pointer_ = nullptr;
```
The previous version was better.